nixos/modules/services/monitoring/prometheus/exporters/ipmi.nix

```diff
@@ -0,0 +1,54 @@
+{ config, lib, pkgs, options }:
+
+with lib;
+
+let
+  logPrefix = "services.prometheus.exporter.ipmi";
+  cfg = config.services.prometheus.exporters.ipmi;
+
+  checkConfigLocation = file:
+    if file == null then
+      null
+    else if lib.hasPrefix "/tmp/" file then
+      throw
+      "${logPrefix}: configuration file must not reside within /tmp - it won't be visible to the systemd service."
+    else
+      file;
+in {
+  port = 9290;
+
+  extraOpts = {
+    configFile = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = lib.mdDoc ''
+        Path to configuration file.
+      '';
+    };
+
+    webConfigFile = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = lib.mdDoc ''
+        Path to configuration file that can enable TLS or authentication.
+      '';
+    };
+  };
+
+  serviceOpts = let
+    checkedConfigFile = checkConfigLocation cfg.configFile;
+    checkedWebConfigFile = checkConfigLocation cfg.webConfigFile;
```

Comment:
I'm not familiar with assertions. I've tried looking at examples of how it's used, but none of them are in similar context. Do you have any pointers on how it can be used in this kind of a module?

Reply:
Ok I think I figured it out, I'll move to assertions.

Reply:
Something along the lines of:
```
assertions = [
  {
    assertion = cfg.configFile != null && lib.hasPrefix "/tmp/" cfg.configFile;
    message = "Configuration file blah blah blah!";
  }
];
```

Reply:
Yeah, I just didn't realize it had to go into the prometheus/exporters.nix and not the ipmi.nix. Works good now.